tokio-trace-serde/src/lib.rs

```diff
@@ -0,0 +1,296 @@
+/// Implements `serde::Serialize` to write `Event` data to a serializer.
+#[derive(Debug)]
+pub struct SerializeEvent<'a>(pub (crate) Event<'a>);
+
+impl<'a> Serialize for SerializeEvent<'a> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where S: Serializer {
+        let serializer = serializer.serialize_struct("Event", 2)?;
+        let mut visitor = SerdeStructVisitor {
+                            serializer,
+                            state: Ok(()),
+                          };
+        self.0.record(&mut visitor);
+        visitor.finish()
+    }
+}
+
+/// A Serde visitor to pull `Attributes` data out of a serialized stream
+#[derive(Debug)]
+pub struct SerializeAttributes<'a>(pub (crate) Attributes<'a>);
+
+impl<'a> Serialize for SerializeAttributes<'a> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where S: Serializer {
+        let mut serializer = serializer.serialize_struct("Attributes", 3)?;
+        serializer.serialize_field("metadata", &SerializeMetadata(self.0.metadata()))?;
+        serializer.serialize_field("parent", &self.0.parent().map(SerializeId))?;
+        serializer.serialize_field("is_root", &self.0.is_root())?;
+
+        let mut visitor = SerdeStructVisitor {
+                            serializer,
+                            state: Ok(()),
+                          };
+        self.0.record(&mut visitor);
+        visitor.finish()
+    }
+}
+
+/// A Serde visitor to pull `Record` data out of a serialized stream
+#[derive(Debug)]
+pub struct RecordValues<'a>(pub (crate) Record<'a>);
+
+impl<'a> Serialize for RecordValues<'a> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where S: Serializer {
+        let serializer = serializer.serialize_map(None)?;
+        let mut visitor = SerdeMapVisitor {
+                            serializer,
+                            state: Ok(()),
+                          };
+        self.0.record(&mut visitor);
+        visitor.finish()
+    }
+}
+
+struct SerdeMapVisitor<S: SerializeMap> {
+    serializer: S,
+    state: Result<(), S::Error>
+}
+
+impl<S> Visit for SerdeMapVisitor<S> where S: SerializeMap {
+    fn record_bool(&mut self, field: &Field, value: bool) {
+        // If previous fields serialized successfully, continue serializing,
+        // otherwise, short-circuit and do nothing.
+        if self.state.is_ok() {
+            self.state = self.serializer
+                .serialize_entry(field.name(), &value)
+        }
+    }
+
+    fn record_debug(&mut self, field: &Field, value: &fmt::Debug) {
+        if self.state.is_ok() {
+            self.state = self.serializer
+                .serialize_entry(field.name(), &format_args!("{:?}", value))
+        }
+    }
+
+    fn record_u64(&mut self, field: &Field, value: u64) {
+        if self.state.is_ok() {
+            self.state = self.serializer
+                .serialize_entry(field.name(), &format_args!("{:?}", value))
```

Comment:
I don't think the serializers for `record_u64`, `record_i64` and `record_str` should use `format_args!`, since `i64`, `u64`, and `str` have their own `Serializable` implementations.
```suggestion
                .serialize_entry(field.name(), &value)
```